modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -253,6 +257,9 @@ public class IgniteImpl implements Ignite {
 
     private final RestAddressReporter restAddressReporter;
 
+    /** CompletableFuture of {@link ClusterRestConfiguration} that will be resolved after the cluster initialization. */
+    private final CompletableFuture<ClusterRestConfiguration> clusterRestConfigurationFuture = new CompletableFuture<>();
```

Comment:
I think that this logic should be encapsulated in a separate class, for example, `DistributedConfigurationUpdater`. This class will know how to apply the AuthConfiguration (which will be provided by the CMG Manager) after all components have started. What do you think?

Reply:
see DistributedConfigurationUpdater